app/controllers/warehouse_controller.rb

```diff
@@ -43,16 +32,17 @@ def create
       company.users << user
       company.warehouses << warehouse
       warehouse.users << user
-      render json: { warehouse: warehouse, admin: user }, status: :created
+      render json: { warehouse: warehouse, admin: user }
     else
       render json: { warehouse_errors: warehouse.errors.full_messages, user_errors: user.errors.full_messages },
              status: :unprocessable_entity
     end
   end
 
   def update
-    if Warehouse.find(params[:id]).update(warehouse_params)
-      render json: { warehouse: Warehouse.find(params[:id]) }, status: :ok
+    warehouse = Warehouse.find(params[:id])
+    if warehouse.update(warehouse_params)
+      render json: warehouse
     else
       render json: { warehouse_errors: Warehouse.find(params[:id]).errors.full_messages }, status: :unprocessable_entity
```

Comment:
i think you should use var warehouse instead finding warehouse second time

Reply:
you're right, just missed it